### ISSUE TITLE
Fix linting issue due to Too Generic Exception Caught

### ISSUE THREAD

**Issue by angrezichatterbox** [scribe-org/Scribe-Android#129]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-Android/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-Android's [Code of Conduct](https://github.com/scribe-org/Scribe-Android/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

The `TooGenericExceptionCaught` rule is currently disabled. This rule helps in detecting overly generic exception handling, which can obscure errors. Re-enabling this rule will help improve error handling practices.

**Steps to see which lines have to be modified:**
1. Comment or set to true the following line in `detekt.yml`:
   ```yaml
   exceptions:
       TooGenericExceptionCaught:
           active: true
   ```
2. Run the command `./gradlew detekt`.
3. Review the results to identify overly generic exception catches that need attention.

### Contribution

I would love to work on this issue and am more than willing to help anyone fix this issue. :)

**Comment by angrezichatterbox** [comment]
I will work on this issue :)

**Comment by andrewtavis** [comment]
Thanks for picking it up, @angrezichatterbox!

**Comment by andrewtavis** [comment]
Closed by #192 😊 Thank you, @angrezichatterbox! Hopefully we'll be able to get rid of the rest of the unused code in the coming month or so :) Let me know if you have ideas for ways to find it so we can make issues for it :)

**Comment by angrezichatterbox** [comment]
> Closed by #192 😊 Thank you, @angrezichatterbox! Hopefully we'll be able to get rid of the rest of the unused code in the coming month or so :) Let me know if you have ideas for ways to find it so we can make issues for it :)

I'll look into the ways to check for unused code or unnecessary code and I'll mention it here or on elements if I find the perfect way to do so.
